cunumeric/deferred.py

```diff
@@ -631,8 +666,12 @@ def _create_indexing_array(
                     # in case of the mixed indises we all nonzero
                     # for the bool array
                     k = k.nonzero()
-                    shift += len(k) - 1
-                    tuple_of_arrays += k
+                    if k[0].size == 0:
+                        store = store.project(dim + shift, 0)
+                        store = store.promote(dim + shift, 0)
```

Comment:
if there is no `true` elements in the boolean array, we will just transform the original array

Reply:
I see the optimization opportunity here, but note that, since `nonzero()` has an unknown output size, the check on `k[0].size` has to block until `nonzero()` returns, even if the call happens to return a non-empty array, and we would rather avoid blocking in the top-level program.

I think the original code here would still work, we would just get an empty indices array out of `_zip_indices`, does that sound right?

Reply:
Yes, original code will work. Removing this special case